downstairs/src/extent.rs

```diff
@@ -688,8 +701,15 @@ pub(crate) fn move_replacement_extent<P: AsRef<Path>>(
                     e
                 );
             }
+            info!(
+                log,
+                "Moved db-shm file {:?} to {:?}",
+                new_file.clone(),
+                original_file.clone()
+            );
             sync_path(&original_file, log)?;
         } else if original_file.exists() {
+            assert!(!clone);
```

Comment:
Add a comment explaining why this is impossible (I think it's because the "original" region is empty?)

Reply:
Yeah, the original region in this case is the "new" region we just created, and that new region better not have
any SQLite files in it.  If it does, then something terrible is about to happen.

Reply:
The new comment is confusing: if we're cloning a region, then the comment suggests that the region is created using `Backend::RawFile`, but then we (possibly) copy SQLite-formated extents into it?

Thinking about it a little more, I guess it's fine because we close the extent after creating it, then reopen it after cloning (and decide dynamically upon reopening whether it's raw or SQLite).  Still, it's a little odd.

I wonder if it would be worth creating a `Backend::Empty`, which creates the directory but no data files (and returns an error for every `ExtentInner` function)?

Reply:
> The new comment is confusing: if we're cloning a region, then the comment suggests that the region is created using Backend::RawFile, but then we (possibly) copy SQLite-formated extents into it?

Yes, the idea here is that we use the regular path (in Omicron) to create a region using whatever the
default standard region backend is, which would be fewer changes on the Omicron side.  Then, the clone
command just ignores whatever backend version the current region is and replaces it with the source.

I think `Backend::Empty` idea is interesting, but it would require updates on the omicron side to create
a region with a different set of options than default.  Not terrible, just more work.